### PR TITLE
Webapp 223 connection skipped modals

### PR DESCRIPTION
**Pull request by federicoweber**
To show the connection prompt modal in Analytics and Engagement, if the user has no channels connected.
Jira tasks: [webapp-223](https://buffer.atlassian.net/browse/WEBAPP-223), [webapp-224](https://buffer.atlassian.net/browse/WEBAPP-224)

image
image

### CODE REVIEW

**Comment by federicoweber** [comment]
hey team,
I need to fix tests, and add a few more before merging, but just wanted to get your eyes on this before I start my day one Monday 🙂 .

Thanks!

**Review by esclapes — APPROVED** [review]
LGTM @federicoweber I really love to see how you use `storybook` for development. 

I am curious to know if this is how you develop in publish too. Would you be up for doing a snack chat on your dev flow?

**Review by MayaUribe — APPROVED** [review]
This looks great @federicoweber 😊

File: packages/app-shell/src/components/Modal/utils.js [hunk 1]
```diff
@@ -18,6 +19,21 @@ export function hasSeenFreeUserStartTrialPrompt() {
   );
 }
 
+export function shouldShowChannelConnectionPrompt(user) {
+  const activeProduct = getActiveProductFromPath();
+  console.log('user.channels.channels', user.currentOrganization.channels);
+  console.log('user.channels.featureFlips', user.currentOrganization.featureFlips);
+  console.log('activeProduct', activeProduct);
```
[the remark this answers is not in the document]
Comment (federicoweber) [reply packages/app-shell/src/components/Modal/utils.js:26]: yep will clean that up once tests are 🟢 🙂

**Comment by federicoweber** [comment]
Hey @esclapes 
> I am curious to know if this is how you develop in publish too. Would you be up for doing a snack chat on your dev flow?
It works well in here because it's all quite isolated, in the Publish codebase it's a bit trickier, but this is one of the reasons I was also pushing to start having components more isolated 😉 .

Also yes, keen to organize a snack chat about my development workflow :)

**Comment by federicoweber** [comment]
hey @bufferapp/core-webapp , I've added both unit and cypress tests in here.
It's admittedly redundant but it was looking like a good chance to show examples for both 🙂.

**Review by amooabeebadesina — APPROVED** [review]

**Pull request merged**